docs/general/administration/hardware-acceleration/apple.md

```diff
@@ -0,0 +1,84 @@
+---
+uid: admin-hardware-acceleration-apple
+title: Apple Mac
+---
+
+# HWA Tutorial On Mac
+
+This tutorial guides you on setting up full video hardware acceleration on Apple Macs via VideoToolbox.
+
+## Acceleration Methods
+
+[VideoToolbox](https://developer.apple.com/documentation/videotoolbox) is the only available method on macOS.
+
+To achieve full acceleration, [Metal](https://developer.apple.com/metal/) and OpenCL are also required.
+
+## Tone-mapping Methods
+
+Hardware accelerated HDR to SDR tone-mapping is supported on all Macs from 2017 and later, with the exception of the MacBook Air (13-inch, 2017).
+
+There are two different methods that can be used. Pros and cons are listed below:
+
+1. **OpenCL**
+
+   - Pros: Supports Dolby Vision P5, detailed fine-tuning options.
+
+   - Cons: Slower, requires more powerful GPUs.
+
+2. **VideoToolbox Native**
+
+   - Pros: Lower power consumption, less dependency on GPU performance, good visual quality without manual fine-tuning.
+
+   - Cons: Lack of tuning options, potential performance issues on Intel Macs, does not support Dolby Vision P5.
+
+When both methods are enabled, VideoToolbox Native will be used for most videos, and OpenCL will only be used as a fallback for Dolby Vision Profile 5 videos.
+
+## Select System Hardware
+
+Hardware accelerated transcoding is supported on all Macs that support [VideoToolbox](https://developer.apple.com/documentation/videotoolbox). This includes most 2011 and later Macs.<sup>*</sup>
+
+Full acceleration is available on most Macs from 2017 and later, with the exception of the MacBook Air (13-inch, 2017).
+
+To have native Apple Silicon support, Jellyfin server 10.9.0+ and `jellyfin-ffmpeg6` 6.0.1-4 or higher is required.
+
+While hardware acceleration via VideoToolbox might work on older series Macs, it is not officially supported.
+
+<sup>*</sup> VideoToolbox may fallback to software decoding/encoding when the task exceeds its hardware capability. The older your Mac, the more this will occur. Macs from 2017 and later are strongly recommended.
```

Comment:
Even if the `allow_sw` option is [disabled by default](https://github.com/FFmpeg/FFmpeg/blob/c31d2326560087b80f8523b27060ad6388369cac/libavcodec/videotoolboxenc.c#L2849), will it automatically fall back to software encoding on partially supported devices?

Reply:
Both encoding and decoding have fallback behaviors. For encoding it will error out, but for decoding, the default option is to [allow software decoding for HEVC videos](https://github.com/FFmpeg/FFmpeg/blob/c31d2326560087b80f8523b27060ad6388369cac/libavcodec/videotoolbox.c#L816).

Reply:
One more question: Is there an artificial limit on the number of concurrent encoding sessions on Apple like NVIDIA NVENC?

Reply:
There is no artificial limit, but there are some hard limits on certain models due to hardware limitations. On some ancient Intel Macs, the limit is only 1, as Intel could not perform concurrent encoding at that time. This should not be an issue on modern Macs anymore; I've run approximately 40 sessions without encountering any problems.

Reply:
Good to know. I remember the Plex documentation mentioned a limit of up to 1 concurrent encoding. It might be worth add this in our documentation to avoid confusion.

Reply:
The problem is, I don't know exactly on which model this was an issue and since when this limitation was no longer applicable.

Reply:
Just put what we already know into the docs. And emphasize that we have only verified this on Apple Silicon, newer Intel based Macs may also support this but use it at your own risk.

This is much better than Plex documents anyway.

> macOS is only capable of hardware-accelerated encoding of 1 video at a time. This is a platform limitation from Apple.

https://support.plex.tv/articles/115002178853-using-hardware-accelerated-streaming/

Reply:
Then I think just change "VideoToolbox may fallback to software decoding/encoding when the task exceeds its hardware capability." to " VideoToolbox may fallback to software decoding/encoding or stop working when the task exceeds its hardware capability." is enough. Users should not care about a historical limitation too much.

Reply:
That should be enough. We will add automatic fallback to software transcoding on the server side in the future.